apps/condo/domains/contact/hooks/useContactExportTaskUIInterface.ts

```diff
@@ -0,0 +1,62 @@
+import get from 'lodash/get'
+
+import { useIntl } from '@open-condo/next/intl'
+
+import { ITask, TASK_REMOVE_STRATEGY } from '@condo/domains/common/components/tasks'
+import { TasksCondoStorage } from '@condo/domains/common/components/tasks/storage/TasksCondoStorage'
+import { TASK_COMPLETED_STATUS } from '@condo/domains/common/constants/tasks'
+import { useDownloadFileFromServer } from '@condo/domains/common/hooks/useDownloadFileFromServer'
+import { ContactExportTask } from '@condo/domains/contact/utils/clientSchema'
+
+import type { ContactExportTask as ContactExportTaskType } from '@app/condo/schema'
+
+interface IUseContactExportTaskUIInterface { (): ({ ContactExportTask: ITask }) }
+
+export const useContactExportTaskUIInterface: IUseContactExportTaskUIInterface = () => {
+    const intl = useIntl()
+    const ContactExportTaskProgressTitle = intl.formatMessage({ id: 'tasks.ContactExportTask.progress.title' })
+    const ContactExportTaskProgressDescriptionPreparing = intl.formatMessage({ id: 'tasks.ContactExportTask.progress.description.preparing' })
+    const ContactExportTaskProgressDescriptionProcessing = intl.formatMessage({ id: 'tasks.ContactExportTask.progress.description.processing' })
+    const ContactExportTaskProgressDescriptionCompleted = intl.formatMessage({ id: 'tasks.ContactExportTask.progress.description.completed' })
+
+    const { downloadFile } = useDownloadFileFromServer()
+
+    const safeDownloadFile = async (taskRecord: ContactExportTaskType) => {
```

Comment:
Why safe?

Reply:
Because if not wrap download file with this function client side app will crush